packages/core/helper-plugin/src/features/Tracking.tsx

```diff
@@ -0,0 +1,321 @@
+import * as React from 'react';
+
+import axios, { AxiosResponse } from 'axios';
+
+import { useAppInfo } from './AppInfo';
+
+export interface TelemetryProperties {
+  useTypescriptOnServer?: boolean;
+  useTypescriptOnAdmin?: boolean;
+  isHostedOnStrapiCloud?: boolean;
+  numberOfAllContentTypes?: number;
+  numberOfComponents?: number;
+  numberOfDynamicZones?: number;
+}
+
+export interface TrackingContextValue {
+  uuid: string | boolean;
+  deviceId?: string;
+  telemetryProperties?: TelemetryProperties;
+}
+
+/* -------------------------------------------------------------------------------------------------
+ * Context
+ * -----------------------------------------------------------------------------------------------*/
+
+const TrackingContext = React.createContext<TrackingContextValue>({
+  uuid: false,
+  deviceId: undefined,
+  telemetryProperties: undefined,
+});
+
+/* -------------------------------------------------------------------------------------------------
+ * Provider
+ * -----------------------------------------------------------------------------------------------*/
+
+export interface TrackingProviderProps {
+  children: React.ReactNode;
+  value?: TrackingContextValue;
+}
+
+const TrackingProvider = ({ value = { uuid: false }, children }: TrackingProviderProps) => {
+  const memoizedValue = React.useMemo(() => value, [value]);
+
+  return <TrackingContext.Provider value={memoizedValue}>{children}</TrackingContext.Provider>;
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * Hook
+ * -----------------------------------------------------------------------------------------------*/
+
+/**
+ * We can group these events together because none have properties so there's no benefit
+ * to having them as separate types.
+ *
+ * Meanwhile those with properties have different property shapes corresponding to the specific
+ * event so understanding which properties go with which event is very helpful.
+ */
+interface EventWithoutProperties {
```

Comment:
1. We can, I think realistically we'd need to document tracking on the whole but trying to add a new event will fail in TS because it won't be part of the union, do you think we should do this now? 🤔 
2. I've done a big sweep mainly looking at for the function `trackUsage` and i've not looked at all the code (because there's a lot) so I might have missed some, which isn't a bad thing we can add as we find them, i'll add the two you've just pointed out 👍🏼